src/components/routes/index.js

```diff
@@ -50,14 +50,17 @@ class Routes extends React.Component<PropsType> {
           <Route path={ROUTE.PATH.SIGN_UP_OAUTH} component={SignUpOAuth} />
           <Route path={ROUTE.PATH.SIGN_UP} component={SignUp} />
           <Route path={ROUTE.PATH.SIGN_UP_STORE} component={SignUpStore} />
+
           {/* Payment */}
           <Route
             path="/payment/:address/:amount/:orderId"
             component={Payment}
           />
 
-          {/* Redirect if logged out */}
-          {!currentUserId && <Redirect to={ROUTE.PATH.HOME} />}
+          {/* Redirect if logged out and not paying*/}
+          {!window.location.href.includes('/payment/') && !currentUserId && (
+            <Redirect to={ROUTE.PATH.HOME} />
+          )}
```

Comment:
Revert. This is redundant.

`<Switch>` goes through its children one-by-one and only renders the first match ignoring everything afterwards, so since this is after the payment route it will not get here if `/payment/:address/:amount/:orderId` have matched and payment was rendered.

Reply:
All the routes below require a user to be signed in so this serves as a barrier for unauthorized users trying to jump in with a direct link